homeassistant/components/homekit/__init__.py

```diff
@@ -124,6 +126,11 @@ def get_accessory(hass, state, aid, config):
     elif state.domain == 'lock':
         a_type = 'Lock'
 
+    elif state.domain == 'media_player':
+        config[CONF_MODE] = validate_media_player_modes(state, config)
+        if config[CONF_MODE] is not None:
```

Comment:
Since you already pass `config` to `validate_media_player_modes`, why not override `config[CONF_MODE]` there? Also the `is not None` check can be abbreviated.
```py
validate_media_player_modes(state, config)
if config[CONF_MODE]:
```